doc/api/cli.md

```diff
@@ -2290,6 +2290,7 @@ This option may be specified multiple times to exclude multiple glob patterns.
 
 If both `--test-coverage-exclude` and `--test-coverage-include` are provided,
 files must meet **both** criteria to be included in the coverage report.
+However, if there is an overlap between the two, `--test-coverage-include` takes precedence.
```

Comment:
I'm not sure if this is related to tests including other tests (that I just agreed to 😅), but I generally expect `exclude` to trump `include`, and there's a fair amount of precedent for that (ex tsconfig include vs exclude).

Reply:
Hey @JakobJingleheimer, thanks for reviewing this 😁 

Regarding this comment: In the other PR (the one currently stale), another contributor proposed a more invasive approach that involved leveraging a list of test files passed to the coverage. One of the doubts about that approach was that it was too invasive and did not leverage the existing implementation.

In this PR, I tried to build upon the existing implementation. The current limitation (if I'm not mistaken) is that, without reversing the order of include/exclude, the user would have no way to add the tests back to the coverage. This is because I am now adding the tests to the exclusion list.

As I mentioned in the PR, I'm not convinced about this approach. Tools like Vitest, for instance, populate exclusion and inclusion lists by default and allow users to override these values. This could be a better solution but would require introducing a "configuration file" to ensure a good developer experience.

Given the current state, a first step could be introducing another flag, such as "--test-coverage-include-tests," to disable the exclusion of test files. This approach might be acceptable since, by default, users should not include test files.

I find this approach more expressive but also dislike the idea of adding additional flags.

What do you think?

Reply:
Another approach could be to ignore by default the files run by the test runner, without allowing the user to add them back. This would remain the case at least until we implement a different management system for the test configurations.

FYI @cjihrig @MoLow

Reply:
Regarding what should trump what with exclude vs. include - I think we should do whatever c8 does here.

I just ran c8 on a project and without specifying any include/exclude options, I see coverage of the test files reported by default. It also looks like exclude trumps include. Please verify that I did that correctly.

I think if we try to use defaults different from ^ we will end up with edge cases and a bunch of extra logic for handling the test files.

Reply:
Hey @cjihrig,  
Yes, I agree. I'll revert the changes made to the include/exclude behavior.  

I’ve been reviewing `c8`, and :  

```
  -n, --include                    A list of specific files that should be
                                   covered (glob patterns are supported)
                                                                   [default: []]
  -x, --exclude                    A list of specific files and directories that
                                   should be excluded from coverage (glob
                                   patterns are supported)             [default:
  ["coverage/**","packages/*/test{,s}/**","**/*.d.ts","test{,s}/**","test{,-*}.{
  js,cjs,mjs,ts,tsx,jsx}","**/*{.,-}test.{js,cjs,mjs,ts,tsx,jsx}","**/__tests__/
  **","**/{ava,babel,nyc}.config.{js,cjs,mjs}","**/jest.config.{js,cjs,mjs,ts}",
         "**/{karma,rollup,webpack}.config.js","**/.{eslint,mocha}rc.{js,cjs}"]]
```

something similar happens also in `Vitest`: https://vitest.dev/config/#coverage-exclude  

I would suggest that we follow the same logic: aka set a default for our `options.coverageExcludeGlobs`.  

I just need to verify whether the user can now override the default with an empty value to remove all exclusions.

Does it make sense to you?

Reply:
note: I just took a look, and, if I'm not mistaken, we won’t have a way to set the exclude to the actual default except by using tricks like `--exclude 'non-existing-file'`

Reply:
> Given the current state, a first step could be introducing another flag, such as "--test-coverage-include-tests," to disable the exclusion of test files.

What is the use-case for it? (I can't think of one)

Reply:
> I just ran c8 on a project and without specifying any include/exclude options, I see coverage of the test files reported by default.

I think I remember that being the case and thinking that was silly (how could that possibly be useful to anyone).

> It also looks like exclude trumps include.

I recall that being the case too.

Reply:
> I just ran c8 on a project and without specifying any include/exclude options, I see coverage of the test files reported by default.

@JakobJingleheimer `c8` has a default list of exclusions : https://github.com/bcoe/c8/blob/main/README.md#cli-options--configuration

Reply:
> > Given the current state, a first step could be introducing another flag, such as "--test-coverage-include-tests," to disable the exclusion of test files.
> 
> What is the use-case for it? (I can't think of one)

IMHO, we should allow users to include or exclude freely, without limitations.  
I agree that adding a flag makes no sense. Honestly, I think we should follow the same logic as other tools: provide a default, overridable list of exclusions and inclusions.  

However, due to our current limitations, if we set a default value, it wouldn’t be possible to set it to `[]` via CLI; we would only be able to override the default value by setting at least one value via the flag.  

Given the ongoing discussion about adding a configuration file, I see two options:  

1. Set a default value that is overridable via `--test-coverage-include` (but not settable to an empty array `[]`), and wait for the configuration file to address the empty array scenario.  
2. Wait for the configuration file to land before implementing this feature.  

Option 1 seems preferable, as it will address the most common use scenario, and the introduction of a configuration file could take time.  

Feedback is welcome before moving forward with the implementation.  

cc @nodejs/test_runner

Reply:
@Llorx, here you can find (partially) the discussion of the implementation